src/autoscaler/api/publicapiserver/public_api_handler.go

```diff
@@ -261,17 +234,84 @@ func (h *PublicApiHandler) DetachScalingPolicy(w http.ResponseWriter, r *http.Re
 
 	// find via the app id the binding -> service instance
 	// default policy? then apply that
-
 	w.WriteHeader(http.StatusOK)
-	_, err = w.Write([]byte("{}"))
+	_, err := w.Write([]byte("{}"))
 	if err != nil {
 		logger.Error(ActionWriteBody, err)
 	}
 }
 
-func proxyRequest(pathFn func() string, call func(url string) (*http.Response, error), w http.ResponseWriter, reqUrl *url.URL, parameters *url.Values, requestDescription string, logger lager.Logger) {
-	aUrl := pathFn()
-	resp, err := call(aUrl)
+// TODO this is a copy of part of the attach ... this should use a common function.
```

Comment:
TODO?

Reply:
removed this todo, i tried the refactor to reuse code and it did not looked cleaner. Comment removed here: 71830fd33